prov/verbs/src/verbs_domain.c

```diff
@@ -177,14 +178,53 @@ static int fi_ibv_domain_bind(struct fid *fid, struct fid *bfid, uint64_t flags)
 	return 0;
 }
 
+static void *fi_ibv_rdm_cm_progress_thread(void *dom)
+{
+	struct fi_ibv_domain *domain =
+		(struct fi_ibv_domain *)dom;
+	struct slist_entry *item, *prev;
+	while (domain->rdm_cm->fi_ibv_rdm_tagged_cm_progress_running) {
+		struct fi_ibv_rdm_ep *ep = NULL;
+		slist_foreach(&domain->ep_list, item, prev) {
+			(void) prev;
+			ep = container_of(item, struct fi_ibv_rdm_ep,
+					  list_entry);
+			if (fi_ibv_rdm_cm_progress(ep)) {
+				VERBS_INFO (FI_LOG_EP_DATA,
+				            "fi_ibv_rdm_cm_progress error\n");
+				abort();
+			}
+		}
+		usleep(domain->rdm_cm->cm_progress_timeout);
```

Comment:
Surely there's a better way to handle progress than spinning with a sleep delay.  Can't this block reading from some sort of event queue?